builtin/logical/transit/policy.go

```diff
@@ -324,11 +468,28 @@ func getPolicy(req *logical.Request, name string) (*Policy, error) {
 		return nil, err
 	}
 
+	persistNeeded := false
 	// Ensure we've moved from Key -> Keys
 	if p.Key != nil && len(p.Key) > 0 {
 		p.migrateKeyToKeysMap()
+		persistNeeded = true
+	}
+
+	// With archiving, past assumptions about the length of the keys map are no longer valid
+	if p.LatestVersion == 0 && len(p.Keys) != 0 {
```

Comment:
Can we add a test that checks the upgrade path from the old format to the new format